Parkduksung/ElevenStreetMvpSample/app/src/main/java/com/exam/elevenstreet/data/source/remote/ProductRemoteDataSource.kt

```diff
@@ -0,0 +1,51 @@
+package com.exam.elevenstreet.data.source.remote
+
+
+import com.exam.elevenstreet.R
+import com.exam.elevenstreet.network.RetrofitInstance
+import com.exam.elevenstreet.network.api.ElevenStreetApi
+import com.exam.elevenstreet.util.App
+import com.example.elevenstreet.ProductResponse
+import com.example.elevenstreet.ProductXmlPullParserHandler
+import java.net.URL
+
+
+class ProductRemoteDataSource : ProductRemoteData {
```

Comment:
```suggestion
class ProductRemoteDataSourceImpl priavte constructor(priavte val elevenStreetApi: ElevenSteetApi) : ProductRemoteDataSource {
```
이렇게 elevenStreetApi객체를 외부에서 주입해주세요.
https://defacto-standard.tistory.com/104
합성, 집약 관계

Reply:
f726027 적용완료하였습니다.

Reply:
합성, 집약 관계에 대해서 든 생각이 컴퓨터는 컴퓨터인데 내장형같이 부품을 변경하지 못하는게 합성, 컴퓨터인데 부품을 변경가능한 외장형 컴퓨터가 집약 관계라고 생각이 들었습니다. 이러함은 내장형 컴퓨터 수명이 10년이라 하면 모든 내장 부품 수명또한 컴퓨터 수명과 같다고 볼 수 있고 외장형 컴퓨터 수명이 10년이면 부품들 또한 수명이 동일한게 아니고 다 다르다고 볼 수 있습니다.  즉, elevenStreetApi객체를 외부에서 주입받으므로 집약관계라고 할 수 있다.

Reply:
맞으면 따봉부탁드립니다.ㅋ